### PR TITLE
Bundling backup install script on build

### PR DESCRIPTION
**Pull request by sfoslund**
- 960d32e: Updating azdo pipeline to use build and test scripts (instead of npm scripts)
- 8cfbc7b: Adding install script download to build scripts, copying it to dist in webpack
- fee2414: Adding logic and test coverage for using fallback script

### CODE REVIEW

**Comment by sfoslund** [comment]
I'll hold off on this until #83 is in and we've released 0.1.2, but wanted to get it out there for review in the meantime.

File: vscode-dotnet-runtime-library/src/Acquisition/InstallScriptAcquisitionWorker.ts [hunk 1]
```diff
@@ -35,6 +39,14 @@ export class InstallScriptAcquisitionWorker implements IInstallScriptAcquisition
             return this.scriptFilePath;
         } catch (error) {
             this.eventStream.post(new DotnetInstallScriptAcquisitionError(error));
+
+            // Try to use fallback install script
+            const fallbackPath = this.getFallbackScriptPath();
+            if (fs.existsSync(fallbackPath)) {
```
Comment (NTaylorMullen) [inline vscode-dotnet-runtime-library/src/Acquisition/InstallScriptAcquisitionWorker.ts:45]: Just for clarity, this wouldn't exist if the extension failed to install right?

File: vscode-dotnet-runtime-library/src/Acquisition/InstallScriptAcquisitionWorker.ts [hunk 2]
```diff
@@ -46,4 +58,8 @@ export class InstallScriptAcquisitionWorker implements IInstallScriptAcquisition
         fs.writeFileSync(filePath, scriptContent);
         fs.chmodSync(filePath, 0o777);
     }
+
+    protected getFallbackScriptPath(): string {
```
Comment (NTaylorMullen) [inline vscode-dotnet-runtime-library/src/Acquisition/InstallScriptAcquisitionWorker.ts:62]: Leave a comment stating that this is for testing purposes.

File: vscode-dotnet-runtime-library/src/Acquisition/InstallScriptAcquisitionWorker.ts (hunk 1 above)
Comment (sfoslund) [reply vscode-dotnet-runtime-library/src/Acquisition/InstallScriptAcquisitionWorker.ts:45]: Yes, if it failed to install, if something went wrong with packaging and the install scripts didn't get included, or in tests.

File: build.ps1 [hunk 3]
```diff
@@ -1,6 +1,16 @@
 $errorColor = "Red"
 $successColor = "Green"
 
+Invoke-WebRequest https://dot.net/v1/dotnet-install.ps1 -OutFile "./vscode-dotnet-runtime-library/install scripts/dotnet-install.ps1"
```
Comment (wli3) [inline build.ps1:4]: add retry here

File: build.sh [hunk 4]
```diff
@@ -2,6 +2,22 @@ RED=`tput setaf 1`
 GREEN=`tput setaf 2`
 NC=`tput sgr0`
 
+echo ""
+echo "----------- Bundling Install Scripts -----------"
+echo ""
+curl https://dot.net/v1/dotnet-install.ps1 -o "./vscode-dotnet-runtime-library/install scripts/dotnet-install.ps1"
```
Comment (wli3) [inline build.sh:8]: so is bash

**Comment by sfoslund** [comment]
Leaving this for now, I will merge it once the 0.1.2 release is out.

**Pull request merged**